pkg/server/instancecleaner/cleaner.go

```diff
@@ -0,0 +1,91 @@
+package instancecleaner
```

Comment:
I would probably place this cleaner inside the `server` package, rather than creating a package all for itself? It's only ever used in the server and splitting it out doesn't get us anything (that I can see)

Reply:
Yeah I was 50/50 on it - moved it into server.